src/databricks/labs/remorph/helpers/recon_config_utils.py

```diff
@@ -98,6 +104,117 @@ def prompt_source(self):
+            filter_type = self._prompts.choice("Select the filter type", filter_types)
+            subset_tables = self._prompts.question(f"Enter the tables(separated by comma) to `{filter_type}`")
+            logger.debug(f"Filter Type: {filter_type}, Tables: {subset_tables}")
+            subset_tables = [f"'{table.strip().upper()}'" for table in subset_tables.split(",")]
+
+            include_list = subset_tables if filter_type == "include" else None
+            exclude_list = subset_tables if filter_type == "exclude" else None
+
+        # Get DataSource
+        data_source = get_data_source(self._source, spark, self._ws, secret_scope)
+        logger.debug(f"Listing tables for `{self._source}` using DataSource")
+        # Get TableRecon config
+        recon_config = data_source.list_tables(
+            catalog_schema_dict.get("src_catalog"), catalog_schema_dict.get("src_schema"), include_list, exclude_list
+        )
+
+        logger.debug(f"Fetched Tables `{', '.join([table.source_name for table in recon_config.tables])}` ")
+
+        # Update the target catalog and schema
+        recon_config.target_catalog = catalog_schema_dict.get("tgt_catalog")
+        recon_config.target_schema = catalog_schema_dict.get("tgt_schema")
+
+        logger.info("Recon Config details are fetched successfully...")
+        logger.debug(f"Recon Config : {recon_config}")
+
+        file_name = f"recon_config_{self._source}_{catalog_schema_dict.get('src_catalog')}.json"
+
+        return recon_config, file_name
+
+    def _save_config_details(self, recon_config_json: str, file_name: str):
+        """
+        Save the config details in a file on Databricks Workspace
+        """
+
+        # Create Installation object
+        installation = Installation(self._ws, "remorph")
+        logger.debug(f"Saving the config details for `{self._source}` in `{file_name}` on Databricks Workspace ")
+        installation.upload(filename=file_name, raw=recon_config_json.encode("utf-8"))
+        ws_file_url = f"{installation.install_folder()}/{file_name}"
+        logger.debug(f"Written `{file_name}` on Databricks Workspace ")
+
+        if self._prompts.confirm(f"Open `{file_name}` config file in the browser?"):
+            webbrowser.open(ws_file_url)
+        logger.info(f"Config `{file_name}` is saved at path: `{ws_file_url}` ")
+        return ws_file_url
+
+    def prompt_and_save_config_details(self):
+        """
+        Prompt for source, target catalog and schema names. Get the table list and save the config details
+        """
+        # Check for Secrets Scope
+        self._confirm_secret_scope()
+        recon_config, recon_config_file = self._prompt_config_details()
+        recon_config_json = json.dumps(recon_config, default=vars, indent=2, sort_keys=True)
+        logger.debug(f"recon_config_json : {recon_config_json}")
+        self._save_config_details(recon_config_json, recon_config_file)
```

Comment:
```suggestion
        self._installation.save(recon_config, filename=recon_config_file)
        if self._prompts.confirm(f"Open config file in the browser?"):
            workspace_url = self._installation.workspace_link(recon_config_file)
            webbrowser.open(workspace_url)
```

**please don't re-invent the wheel.** Also, inject `self._installation` via constructor, so that you can use the `MockInstallation`: https://github.com/databrickslabs/blueprint?tab=readme-ov-file#unit-testing-installation-state

Reply:
Sure, will inject `self._installation` via constructor. 
But we couldn't use `installation.save` due to the reasons mentioned [here](https://github.com/databrickslabs/remorph/pull/249#issuecomment-2080688538)